src/core/Svg.tsx

```diff
@@ -0,0 +1,100 @@
+import { Object3DProps, useLoader } from '@react-three/fiber'
+import * as React from 'react'
+import { forwardRef, Fragment, useMemo } from 'react'
+import { Color, DoubleSide, MeshBasicMaterial, Object3D } from 'three'
+import { SVGLoader } from 'three-stdlib'
+
+export interface SvgProps extends Object3DProps {
+  /** src can be a URL or SVG data */
+  src: string
+  skipFill?: boolean
+  skipStrokes?: boolean
+  strokesWireframe?: boolean
+  fillWireframe?: boolean
+}
+
+export const Svg = forwardRef<Object3D, SvgProps>(function R3FSvg(
+  { src, skipFill, skipStrokes, strokesWireframe, fillWireframe, ...props },
+  ref
+) {
+  const svg = useLoader(SVGLoader, !src.startsWith('<svg') ? src : `data:image/svg+xml;utf8,${src}`)
+
+  const shapeGroups = useMemo(
+    () => (skipFill ? [] : svg.paths.map((path) => SVGLoader.createShapes(path))),
+    [svg, skipFill]
+  )
+
+  const fillMaterials = useMemo(
+    () =>
+      skipFill
+        ? []
+        : svg.paths.map(
+            (path) =>
+              new MeshBasicMaterial({
+                color: new Color()
+                  .setStyle(path.userData!.style.fill === 'none' ? '#000000' : path.userData!.style.fill)
+                  .convertSRGBToLinear(),
+                opacity: path.userData!.style.fillOpacity,
+                transparent: true,
+                side: DoubleSide,
+                depthWrite: false,
+                wireframe: fillWireframe,
+              })
```

Comment:
Maybe prefer to map this into JSX. This can leak memory when updated.

Reply:
@CodyJasonBennett the problem with using JSX is that at the the shapes may be sharing the same material, so keeping them in JSX will mean that we will be creating more materials than needed.

`SVGLoader.createShapes( path )` can return (and it does in some of the example files) more than one shape per path/material.

As an alternative, I can add a `useEffect` return function to dispose old materials, so you may still have the best performance and we fix the memory leak.

Let me know what you think.

Reply:
Ok, updated. All the rendering issues on some of the SVG files are now gone and... is so much nicer to read in JSX!

The only downside is that now every shape uses its own material, but I guess that's ok?

The stroke geometry I don't think is possible to convert to JSX, so instead I'm disposing it with `useEffect`.

If you are worried about the missing opportunity of reusing materials, I can apply the same strategy as with the stroke geometry and dispose it with JSX.

Reply:
After testing all the three.js sample shapes, the following have issues, but I don't think is something to do with this PR and instead is because `three-stdlib` `SVGLoader` is out of date compared to the one from `three` `examples` folder.

- Energy: throws error `Uncaught TypeError: Cannot read properties of null (reading 'boundingSphere')
    at Frustum.intersectsObject`
- Style CSS inside def: should be yellow but is black
- Round join: the geometry seems off

Reply:
I can update three-stdlib and you can pump it in this PR if that unblocks you here.

Reply:
It doesn't block this PR, it will just make the SVG loader support more SVG formats. So we can merge this anyway and when `three-stdlib` is updated and `drei` bumps the latest version things should just work.